### PR TITLE
[SPARK-26771][SQL][FOLLOWUP] Make all the uncache operations non-blocking by default

### PR DESCRIPTION
**Pull request by maropu**
## What changes were proposed in this pull request?
To make the blocking behaviour consistent, this pr made catalog table/view `uncacheQuery` non-blocking by default. If this pr merged, all the behaviours in spark are non-blocking by default.

## How was this patch tested?
Pass Jenkins.

### CODE REVIEW

**Comment by maropu** [comment]
@srowen I've checked that all the other default behaviours are non-blocking except for tests.

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 1]
```diff
@@ -365,7 +365,7 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
   override def dropTempView(viewName: String): Boolean = {
     sparkSession.sessionState.catalog.getTempView(viewName).exists { viewDef =>
       sparkSession.sharedState.cacheManager.uncacheQuery(
-        sparkSession, viewDef, cascade = false, blocking = true)
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala]: Let's just take out this argument, and ultimately make a non-blocking call.

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala]: ah, ok.

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala]: I just noticed that `uncacheQuery` has `blocking = true` by default, so I changed it then took out the argument here.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala [hunk 2]
```diff
@@ -65,7 +65,7 @@ case class CachedRDDBuilder(
     _cachedColumnBuffers
   }
 
-  def clearCache(blocking: Boolean = true): Unit = {
+  def clearCache(blocking: Boolean = false): Unit = {
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala]: One more, I changed this into non-blocking, too.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala [hunk 3]
```diff
@@ -103,7 +103,7 @@ class CacheManager extends Logging {
   def uncacheQuery(
       query: Dataset[_],
       cascade: Boolean,
-      blocking: Boolean = true): Unit = {
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: Why even keep the parameter? This is internal to Spark right?

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: Dropped it

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: I'm checking now and just a sec.

Comment (maropu) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala]: ok, removed it.

**Comment by maropu** [comment]
retest this please

**Comment by rezasafi** [comment]
LGTM pending tests. Thank you for the quick work here.

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 4]
```diff
@@ -2956,7 +2956,8 @@ class Dataset[T] private[sql](
    * @since 1.6.0
    */
   def unpersist(blocking: Boolean): this.type = {
-    sparkSession.sharedState.cacheManager.uncacheQuery(this, cascade = false, blocking)
+    sparkSession.sharedState.cacheManager.uncacheQuery(
+      sparkSession, logicalPlan, cascade = false, blocking)
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: I see, this method needs to retain a blocking param to still support this method. OK

File: sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala [hunk 5]
```diff
@@ -93,13 +93,24 @@ class CachedTableSuite extends QueryTest with SQLTestUtils with SharedSQLContext
     }.sum
   }
 
+  // Blocking uncache table for tests
+  private def uncacheTable(tableName: String): Unit = {
```
Comment (srowen) [inline sql/core/src/test/scala/org/apache/spark/sql/CachedTableSuite.scala]: My only concern is whether a user would ever need this behavior as well, but I think it's pretty specific to our tests. I'm OK with it

**Review by srowen — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master.

**Comment by gatorsmile** [comment]
@maropu Could you update the migration guide in SQL doc?

**Comment by maropu** [comment]
ok

**Comment by srowen** [comment]
I added another sentence to the existing release notes for this JIRA. I don't think it requires any migration; there is no API or behavior change that should matter to users. The caching is internal.

**Comment by maropu** [comment]
oh, yes. The release note looks reasonable to me.

**Comment by maropu** [comment]
btw, we still have a room to provide a SQL option (e.g., `spark.sql.cache.blocking`) for changing this default behaivour? I'm a bit worried about the downside of memory pressure as you pointed out before. how about adding this option and writing a doc for that in https://spark.apache.org/docs/2.4.0/sql-performance-tuning.html#caching-data-in-memory?

**Comment by srowen** [comment]
Users can still control the blocking behavior for the Dataset/RDD/Broadcast objects, so I don't think it's worth a new config to change the default. I am also not sure if there's a reasonable scenario where someone needs the internal cache unpersist to be blocking? if memory pressure is an issue, lots of other things are more important than this. I would rather not add yet another random config for this.

**Comment by maropu** [comment]
ok

**Pull request closed without merging**